pandas/tests/reductions/test_stat_reductions.py

```diff
@@ -132,13 +133,21 @@ def _check_stat_op(
                 exp = alternate(s)
                 assert res == exp
 
+            # Series.median(Series('abc'))
             # check on string data
+            msg = (
+                "could not convert string to float|"
+                r"complex\(\) arg is a malformed string|"
+                "can't multiply sequence by non-int of type 'str'|"
+                "Could not convert abc to numeric"
```

Comment:
I think those are too much varied, and not all under our control, to test them. I don't think it is needed (or worth it) here to test the exact error message in this case

Reply:
Good point about them not being all under our control - I just checked and only prod has this issue - "can't multiply sequence by non-int of type 'str'|" is a message from numpy.

But I don't want to completely ignore all the error messages. What would you think `if` I put some if statements in here? there are only really three cases - prod, for which I can use `tm.external_error_raised`, mean, which has one error message "Could not convert abc to numeric", and the rest which have the same error message "could not convert string to float"?

(I don't know where the complex one comes from - I must have made a mistake there)

I will try making that change and see if you like it.

Reply:
Ok I added some logic - let me know what you think.

Reply:
Personally, I think this only makes it more complex, not less ..

To take a step back, the *reason* we are adding those message matches, should not be to have "all of them matched" so we can cross off that one of the list. IMO, we should only do what is useful to test: the cases where we have a specific informative error message, and then test this to ensure we keep it that way. 
In this case, though, it's actually a set of different somewhat accidental error messages that are not very specific. IMO it's only making our tests more complex to work with to add specific asserts for it.  I think it is perfectly fine to sometimes do a "match all".

Reply:
You could also check some simplified subset like `"[C|c]ould not convert|can't multiply"`, which is IMO sufficient

Reply:
Ok I changed it to `None` so it matches anything. Maybe in the future if we get more deliberate about what kind of error message gets returned we can add a more specific match.